### PR TITLE
DEP: special: improve deprecation of private modules named without underscore prefix

### PR DESCRIPTION
**Pull request by j-bowhay**
#### Reference issue

towards https://github.com/scipy/scipy/issues/18279
#### What does this implement/fix?

Currently, some deprecation warnings for private modules named without underscore prefixes are misleading. For example:
Before this pr:
```
In [1]: from scipy.special.orthogonal import arange
<ipython-input-1-18b058905c34>:1: DeprecationWarning: Please use `arange` from the `scipy.special` namespace, the `scipy.special.orthogonal` namespace is deprecated.
  from scipy.special.orthogonal import arange
```
After:
```
In [4]: from scipy.special.orthogonal import arange
<ipython-input-4-18b058905c34>:1: DeprecationWarning: `scipy.special.orthogonal.arange` is deprecated along with the `scipy.special.orthogonal` namespace. `scipy.special.orthogonal.arange` will be removed in SciPy 1.13.0, and the `scipy.special.orthogonal` namespace will be removed in SciPy 2.0.0.
  from scipy.special.orthogonal import arange
```

### CODE REVIEW

File: scipy/special/orthogonal.py [hunk 1]
```diff
@@ -43,13 +42,6 @@ def __dir__():
...
-            f"{name}. Try looking in scipy.special instead.")
-
-    warnings.warn(f"Please use `{name}` from the `scipy.special` namespace, "
-                  "the `scipy.special.orthogonal` namespace is deprecated.",
-                  category=DeprecationWarning, stacklevel=2)
-
-    return getattr(_orthogonal, name)
+    return _sub_module_deprecation(sub_package="special", module="orthogonal",
+                            private_modules=["_orthogonal"], all=__all__,
+                            attribute=name)
```
Comment (lucascolley) [inline scipy/special/orthogonal.py:47]: mind the no-new-line here

File: scipy/special/sf_error.py [hunk 2]
```diff
@@ -16,13 +15,6 @@ def __dir__():
...
-            f"{name}. Try looking in scipy.special instead.")
-
-    warnings.warn(f"Please use `{name}` from the `scipy.special` namespace, "
-                  "the `scipy.special.sf_error` namespace is deprecated.",
-                  category=DeprecationWarning, stacklevel=2)
-
-    return getattr(_sf_error, name)
+    return _sub_module_deprecation(sub_package="special", module="sf_error",
+                            private_modules=["_sf_error"], all=__all__,
+                            attribute=name)
```
Comment (lucascolley) [inline scipy/special/sf_error.py:20]: no new line here either

File: scipy/special/basic.py [hunk 3]
```diff
@@ -85,13 +82,6 @@ def __dir__():
...
-            f"{name}. Try looking in scipy.special instead.")
-
-    warnings.warn(f"Please use `{name}` from the `scipy.special` namespace, "
-                  "the `scipy.special.basic` namespace is deprecated.",
-                  category=DeprecationWarning, stacklevel=2)
-
-    return getattr(_basic, name)
+    return _sub_module_deprecation(sub_package="special", module="basic",
+                            private_modules=["_basic", "_ufuncs"], all=__all__,
+                            attribute=name)
```
Comment (lucascolley) [inline scipy/special/basic.py:87]: Just a style nit but I think you're fine to indent like this and stay within the 88 char limit. As far as I have looked all of the other uses of `_sub_module_deprecation` have indented like this.
```suggestion
    return _sub_module_deprecation(sub_package="special", module="basic",
                                   private_modules=["_basic", "_ufuncs"], all=__all__,
                                   attribute=name)
```

**Review by h-vetinari — COMMENTED** [review]
Approved module the style nits that were pointed out (i.e. once those are fixed, consider it good to go). Thanks!

File: scipy/special/orthogonal.py (hunk 1 above)
Comment (j-bowhay) [inline scipy/special/orthogonal.py:47]:
```suggestion
                            attribute=name)

```

File: scipy/special/sf_error.py (hunk 2 above)
Comment (j-bowhay) [inline scipy/special/sf_error.py:20]:
```suggestion
                            attribute=name)

```

**Review by lucascolley — CHANGES_REQUESTED** [review]
The indentation change can be applied to all of the files, not just the one fixed above. Otherwise LGTM

**Review by lucascolley — APPROVED** [review]

**Pull request merged**